File: HaffmanEncode.java
import java.util.ArrayList;
import java.util.Arrays;

class Node{
    int key, freq, left, right;
    Node(){
        freq=0;
    };
    Node(int key,int freq,int left,int right){
        this.key=key;
        this.freq=freq;
        this.left=left;
        this.right=right;
    }
}

class S{
    char c;
    int freq;
    public S(char c,int freq){
        this.c = c;
        this.freq = freq;
    }
    @Override
    public String toString() {
        return c + " " + freq;
    }
}
class HaffmanEncode{
    public static void main(String[] args){
        char letter[]={'S','H','U','N','S','U','K','E','O','N','U','K','I'};
        ArrayList<S> str = new ArrayList<>();

        Arrays.sort(letter);

        for(char l : letter){
            boolean frag = false;
            if(str.isEmpty()){
                str.add(new S(l,1));
                continue;
            }
            for(S s:str){
                if(s.c == l){
                    s.freq++;
                    frag = true;
                }
            }
            if(!frag)str.add(new S(l,1));
        }



        System.out.println(str);
        
    }
}

File: HaffmanEncode.cpp
#include<iostream>
using namespace std;

struct Node{
    int key;
    Node *right, *left, *parent;
};

struct Letter{
    char C[13]={'S','H','U','N','S','U','K','E','O','N','U','K','I'};
    int size = 13;
} ;

Node *root, *NIL;

int main(){
    char C[]={'S','H','U','N','S','U','K','E','O','N','U','K','I'};
    int n = C.size;
Q = priority_queue()
for i = 1 to n
    n = node(C[i])
    Q.push(n)
end for
while Q.size() is not equal to 1
    Z = new node()
    Z.left = x = Q.pop
    Z.right = y = Q.pop
    Z.frequency = x.frequency + y.frequency
    Q.push(Z)
end while
Return Q
    return 0;
}